tartufo/scanner.py

```diff
@@ -461,22 +461,36 @@ def chunks(self) -> Generator[types.Chunk, None, None]:
         """
         already_searched: Set[bytes] = set()
 
-        try:
-            if self.git_options.branch:
-                branches = self._repo.remotes.origin.fetch(self.git_options.branch)
-            else:
-                branches = self._repo.remotes.origin.fetch()
-        except git.GitCommandError as exc:
-            raise types.GitRemoteException(exc.stderr.strip()) from exc
-
-        for remote_branch in branches:
+        if self.git_options.branch:
+            # Single branch only
+            unfiltered_branches = list(self._repo.branches)
+            branches = [
+                x for x in unfiltered_branches if x.name == self.git_options.branch
+            ]
+            try:
+                if self.git_options.fetch:
+                    print("Fetching remote origin/" + self.git_options.branch)
+                    self._repo.remotes.origin.fetch(self.git_options.branch)
+            except git.GitCommandError as exc:
+                raise types.GitRemoteException(exc.stderr.strip()) from exc
+        else:
+            # Everything
+            branches = list(self._repo.branches)
+            try:
+                if self.git_options.fetch:
+                    print("Fetching remote origin (all branches)")
+                    self._repo.remotes.origin.fetch()
+            except git.GitCommandError as exc:
+                raise types.GitRemoteException(exc.stderr.strip()) from exc
+
+        print("Scanning " + str(len(branches)) + " branches")
+        for branch in branches:
```

Comment:
This whole section looks to me like it could be condensed back down to the original form and eliminate a whole lot of duplicate code. Unless I'm missing something?

Also, the print calls have got to go!